integration-tests/src/test/java/oracle/kubernetes/operator/ItManagedCoherence.java

```diff
@@ -0,0 +1,338 @@
+                domainMap.put("voyagerWebPort", new Integer("30366"));
+            }
+            domain = TestUtils.createDomain(domainMap);
+            domain.verifyDomainCreated();
+            String[] pods = {
+                    DOMAINUID1 + "-" + domain.getAdminServerName(),
+                    DOMAINUID1 + "-managed-server",
+                    DOMAINUID1 + "-managed-server1",
+                    DOMAINUID1 + "-managed-server2",
+                    DOMAINUID1 + "-new-managed-server1",
+                    DOMAINUID1 + "-new-managed-server2",
+            };
+            verifyServersStatus(domain, pods, DOMAINUID1);
+
+            // Build WAR in the admin pod and deploy it from the admin pod to a weblogic target
+            TestUtils.buildDeployCoherenceAppInPod(domain,
+                    testAppName, scriptName, BaseTest.getUsername(), BaseTest.getPassword(),
+                    appToDeploy, "dataCluster");
+
+            coherenceCacheTest();
+
+            testCompletedSuccessfully = true;
+        } finally {
+            if (domain != null && !SMOKETEST && (JENKINS || testCompletedSuccessfully)) {
+              domain.destroy();
+            }
+        }
+        logger.info("SUCCESS - " + testMethodName);
+    }
+    /**
+     * Verifies all of the servers in the cluster are in Running status.
+     *
+     * @param domain Domain
+     * @param pods   array pod names to check the status for
+     */
+    private static void verifyServersStatus(Domain domain, String[] pods, String domainUID) {
+        K8sTestUtils testUtil = new K8sTestUtils();
+        String domain1LabelSelector = String.format("weblogic.domainUID in (%s)", domainUID);
+        String namespace = domain.getDomainNs();
+        for (String pod : pods) {
+            assertTrue(
+                    pod + " Pod not running", testUtil.isPodRunning(namespace, domain1LabelSelector, pod));
+        }
+    }
+
+    private void coherenceCacheTest() throws Exception {
+
+        String[] firstNameList = {"Frodo", "Samwise", "Bilbo", "peregrin", "Meriadoc", "Gandalf"};
+        String[] secondNameList = {"Baggins", "Gamgee", "Baggins", "Took", "Brandybuck", "TheGrey"};
+        ExecResult result;
+
+        for (int i = 0; i < firstNameList.length; i++) {
+           result = addDataToCache(firstNameList[i], secondNameList[i]);
+           logger.info("addDataToCache returned" + result.stdout());
+        }
+        //check if cache size is 6
+        result = getCacheSize();
+        logger.info("number of records in cache = " + result.stdout());
+        if (!(result.stdout().equals("6"))) {
+            logger.info("number of records in cache = " + result.stdout());
+            assertFalse("Expected 6 records", "6".equals(result.stdout()));
+        }
+        //get the data from cache
+        result = getCacheContents();
+        logger.info("Cache contains the following entries \n" + result.stdout());
```

Comment:
You should validate the cache contents.  Ideally this should be done in the app running in WebLogic, there is really no reason to transfer everything back to the client.